enterprise/bloom-index/src/main/java/org/neo4j/kernel/api/impl/bloom/BloomLuceneIndex.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.impl.bloom;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.store.Directory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.neo4j.helpers.TaskCoordinator;
+import org.neo4j.kernel.api.impl.index.AbstractLuceneIndex;
+import org.neo4j.kernel.api.impl.index.partition.AbstractIndexPartition;
+import org.neo4j.kernel.api.impl.index.partition.IndexPartitionFactory;
+import org.neo4j.kernel.api.impl.index.partition.PartitionSearcher;
+import org.neo4j.kernel.api.impl.index.storage.PartitionedIndexStorage;
+
+import static java.util.Collections.singletonMap;
+
+class BloomLuceneIndex extends AbstractLuceneIndex
+{
+
+    private String[] properties;
+    private final Analyzer analyzer;
+
+    BloomLuceneIndex( PartitionedIndexStorage indexStorage, IndexPartitionFactory partitionFactory, String[] properties, Analyzer analyzer )
+    {
+        super( indexStorage, partitionFactory );
+        this.properties = properties;
+        this.analyzer = analyzer;
+    }
+
+    private static final String KEY_STATUS = "status";
+    private static final String ONLINE = "online";
+    private static final Map<String,String> ONLINE_COMMIT_USER_DATA = singletonMap( KEY_STATUS, ONLINE );
+
+    private final TaskCoordinator taskCoordinator = new TaskCoordinator( 10, TimeUnit.MILLISECONDS );
+
+    public PartitionedInsightBloomWriter getIndexWriter( WritableDatabaseBloomIndex writableIndex ) throws IOException
```

Comment:
The structure here is a bit (maybe unnecessarily?) confusing.

- `BloomLuceneIndex` is an `AbstractLuceneIndex`
- `WritableDatabaseBloomIndex` act as a wrapper for `BloomLuceneIndex` to present it as a `WritableAbstractDatabaseIndex` (subclass of `AbstractLuceneIndex`).
- But this method (`getIndexWriter`) in `BloomLuceneIndex` take `WritableDatabaseBloomIndex` as input. It takes it's own wrapper class as input.

Reply:
Sadly, I think that this is the way it is meant to be, at least for now.